datasketch/hnsw.py

```diff
@@ -489,9 +489,13 @@ def insert(
         if level is None:
             level = int(-np.log(self._random.random_sample()) * self._level_mult)
         self._nodes[key] = _Node(key, new_point)
-        if (
-            self._entry_point is not None
-        ):  # The HNSW is not empty, we have an entry point
+
+        for _ in range(len(self._graphs), level + 1):
+            self._graphs.append(self._layer_class(key))
+
+        if self._entry_point is None:
+            self._entry_point = key
```

Comment:
This seems to indicate that the first point in the index becomes the entry point. I think the entry point should be updated for each new, higher level created. No?

Reply:
Indeed, that is wrong. I will pursue my investigations.